cql3/type_json.cc

```diff
@@ -187,7 +187,63 @@ struct from_json_object_visitor {
         }
         return t.decompose(T(to_int64_t(value)));
     }
-    bytes operator()(const string_type_impl& t) { return t.from_string(rjson::to_string_view(value)); }
+    bytes operator()(const string_type_impl& t) {
+        if (value.IsString()) {
+            return t.from_string(rjson::to_string_view(value)); 
+        } else if (value.IsBool()) {
+            return t.from_string(value.IsTrue() ? "true" : "false");
+        } else if (value.IsNumber()) {
+            sstring out;
+
+            // Not using t.from_string(rjson::print(value))
+            // as it turns out it's less precise for floating point numbers,
+            // for example:
+            //
+            // value:                333834658736487.7474
+            // rjson::print(value):  333834658736487.78
+            // The solution below:   333834658736487.75
+
+            // Not using to_int64_t() helper and manually checking all
+            // possible integer types to correctly handle
+            // the maximum value of uint64_t.
+            if (value.IsInt64()) {
+                out = fmt::format("{}", value.GetInt64());
+            } else if (value.IsInt()) {
+                out = fmt::format("{}", value.GetInt());
+            } else if (value.IsUint()) {
+                out = fmt::format("{}", value.GetUint());
+            } else if (value.IsUint64()) {
+                out = fmt::format("{}", value.GetUint64());
+            } else if (value.IsDouble()) {
+                // Formatting style of JsonCpp, which
+                // cannot be easily replicated without printf.
```

Comment:
Did any particular user also rely on that? I'm asking because now, on the other hand, people might already have started to rely on the rjson output, and we'll never stop this back-and-forth. And a huge advantage of the rjson way is that it doesn't take over 20 lines of careful char processing and pointer arithmetic to get the desired result. Unless there's a good reason to restore the previous format, I'd prefer to avoid open-coding anything.

Reply:
This code handles such a case:
```
INSERT INTO t JSON '{"string_column": 12.34}';
```

and this doesn't work currently (with rjson), so no one is relying on rjson string formatting of floating-point numbers.

Reply:
However, I'm not aware of anyone relying on old formatting of floating point numbers. Therefore `t.from_string(rjson::print(value))` would be OK to use here instead of manually serializing to string here, but there's some problem with worse precision when using `rjson::print`.

Reply:
Fixed in amended commit by just using `rjson::print`